### PR TITLE
Added tests for oneAPI kNN

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

File: cpp/oneapi/dal/algo/knn/BUILD [hunk 1]
```diff
@@ -16,6 +16,34 @@ dal_module(
...
+
+dal_test_suite(
+    name = "tests",
+    host_tests = [
+        ":common_tests",
+    ],
+    dpc_tests = [
+        ":gpu_tests_dpc",
+    ],
+)
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/knn/BUILD:49]: Please fix end of file

File: cpp/oneapi/dal/algo/knn/common_knn_test.cpp [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    constexpr std::int64_t row_count_train = 6;
+
+    const float y_train[] = { 0.f, 0.f, 0.f, 1.f, 1.f, 1.f };
+
+    dal::homogen_table x_train_table;
+    const auto y_train_table = dal::homogen_table::wrap(y_train, row_count_train, 1);
+
+    ASSERT_THROW(
+        dal::train(nn::descriptor<float, nn::method::kd_tree, nn::task::classification>{ 2, 1 }
+                       .set_class_count(0xFFFFFFFF),
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/knn/common_knn_test.cpp:75]: Is it signed or unsigned?

**Review by KulikovNikita — CHANGES_REQUESTED** [review]

File: cpp/oneapi/dal/algo/knn/common_knn_test.cpp (hunk 2 above)
Comment (amgrigoriev) [reply cpp/oneapi/dal/algo/knn/common_knn_test.cpp:75]: std::int64

**Review by KulikovNikita — APPROVED** [review]

File: cpp/oneapi/dal/algo/knn/common_knn_dpc_test.cpp [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+#include "gtest/gtest.h"
+#include "oneapi/dal/algo/knn/infer.hpp"
+#include "oneapi/dal/algo/knn/train.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+using namespace oneapi;
+namespace nn = oneapi::dal::knn;
+
+TEST(knn_overflow_tests, classes_brute_force) {
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/knn/common_knn_dpc_test.cpp:25]: General comment for all tests cases: need better naming

Read this as an example: https://dzone.com/articles/7-popular-unit-test-naming

Possible naming in this case: throws_if_class_count_leads_to_overflow

**Review by rlnx — COMMENTED** [review]
- Is it possible to have universal test cases for brutforce and kd-tree?
- Better naming for test cases is required

File: cpp/oneapi/dal/algo/knn/common_knn_dpc_test.cpp (hunk 3 above)
Comment (amgrigoriev) [reply cpp/oneapi/dal/algo/knn/common_knn_dpc_test.cpp]: "Is it possible to have universal test cases for brutforce and kd-tree?" Do you mean that it's not necessary to test bad args for all methods? Also, Brute Force tests are expected to run on GPU while K-D tree ones - on CPU

**Pull request merged**